### PR TITLE
KK-1065 | fix: fix "yarn generate:graphql" by using graphql-codegen

### PR DESCRIPTION
**Pull request by karisal-anders**
### fix: fix "yarn generate:graphql" by using graphql-codegen

now "yarn lint", "yarn typecheck", "yarn test" and
"yarn generate:graphql" all work and pass

some files were left over with broken types, disabled typechecking for
those files and marked them with FIXME comment

refs KK-1065

## Context

[KK-1065](https://helsinkisolutionoffice.atlassian.net/browse/KK-1065)

## Screenshots

[KK-1065]: https://helsinkisolutionoffice.atlassian.net/browse/KK-1065?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr514.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr514.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr514.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr514.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by karisal-anders** [comment]
I'll refactor a bit so the types are easier to manage.

**Review by nikomakela — APPROVED** [review]
Good job. Some minor notes added.

File: src/domain/event/partial/__tests__/OccurrenceInfo.test.tsx [hunk 1]
```diff
@@ -2,12 +2,27 @@ import React from 'react';
...
+    NonNullable<
+      NonNullable<
+        NonNullable<
+          ChildByIdQuery['child']
+        >['activeInternalAndTicketSystemEnrolments']
+      >['edges'][number]
+    >['node'],
+    'EnrolmentNode'
+  >['occurrence']
+>;
```
[the remark this answers is not in the document]
Comment (karisal-anders) [reply src/domain/event/partial/__tests__/OccurrenceInfo.test.tsx:23]: Yeah, that a bit monstrous perhaps with the nested NonNullables etc. I'll refactor the types a bit in another PR.

[the remark this answers is not in the document]
Comment (karisal-anders) [reply src/domain/event/partial/__tests__/OccurrenceInfo.test.tsx:23]: Refactoring in https://github.com/City-of-Helsinki/kukkuu-ui/pull/515

For example
```typescript
type ChildByIdQueryEnrolmentNodeOccurrence = NonNullable<
  TypeByTypename<
    NonNullable<
      NonNullable<
        NonNullable<
          ChildByIdQuery['child']
        >['activeInternalAndTicketSystemEnrolments']
      >['edges'][number]
    >['node'],
    'EnrolmentNode'
  >['occurrence']
>;
```
→
```
export type InternalEnrolmentOccurrence = NonNullable<
  InternalEnrolment['occurrence']
>;
```
in [ChildByIdQueryTypes.ts](https://github.com/City-of-Helsinki/kukkuu-ui/pull/515/files#diff-a2f2a86659d83b9832f814537023224fbe3dacaf7c06ea69719fc68605186965)

**Pull request merged**